### ISSUE TITLE
Storage plugin saves error message at intended file location when key does not exist

### ISSUE THREAD

**Issue by onlybakam** [aws-amplify/amplify-ios#285]
try to download an object that does not exist. Storage saves the error message at the provide local file location.

```swift
        _ = Amplify.Storage.downloadFile(key: fileKey, local: localURL, options: options) { (event) in
            switch event {
            case .completed:
                break
            case .failed(let storageError):
                print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
            case .inProcess(let progress):
                //print("Download progress -> \(progress)")
                break
            default:
                break
            }
        }
```

`fileKey` does not exist. error message is saved at `localURL`

content of file

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>protected/REDACTED/25E2D406-E361-41F6-952C-B0424E12C0E8.jpeg</Key><RequestId>REDACTED</RequestId><HostId>REDACTED</HostId></Error>
```

**Comment by drochetti** [comment]
@lawmicha do you know if this is still an issue?

**Comment by lawmicha** [comment]
Hi @onlybakam, thanks for reporting this behavior. I was able to reproduce this. 

- Currently the Storage plugin leverages TransferUtility to perform the download to file operation. Related code in TransferUtility [NSURLSessionDownloadDelegate](https://github.com/aws-amplify/aws-sdk-ios/blob/main/AWSS3/AWSS3TransferUtility.m#L2535-L2538) so the file already contains the content from the S3 response on the callback. We can take this as an enchancement to change the behavior of this call
- The error i got was `.keyNotFound` when this fails. I would like to understand the impact of this behavior on your use case, how does it affect your users in the app? is deleting the file when you get `.keyNotFound` error a feasible workaround for you?

```
case .failure(let storageError):
                    print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
                    if case .keyNotFound = storageError {
                        // Delete the file if download fails with key not found
                        try? FileManager.default.removeItem(at: downloadToFileName)
                    }
```

**Comment by onlybakam** [comment]
this was a while back so cant remember if the status was `.failure`.

the `event` and `storageError` is specific so the user can easily handle the situation... i just find it weird that we would then ask the user to clean this up. at the very least we should make this clear in the docs.

impact?
minor. not user friendly.

**Comment by lawmicha** [comment]
This has been released in 1.0.5, thanks for reporting this issue! https://github.com/aws-amplify/amplify-ios/releases/tag/v1.0.5
